compiler/rustc_data_structures/src/sync/worker_local.rs

```diff
@@ -1,4 +1,4 @@
-use crate::sync::Lock;
+use parking_lot::Mutex;
```

Comment:
Why the `Lock` to `Mutex` changes in this file? Is this code not used in the serial front-end?

Reply:
It's not used in the serial compiler and it's quite cold.